pkg/controller/controller.go

```diff
@@ -514,10 +514,9 @@ func getClass(ingressType string) string {
 func makeAnnotations(function *faasv1.FunctionIngress) map[string]string {
 	class := getClass(function.Spec.IngressType)
 	specJSON, _ := json.Marshal(function)
-	annotations := map[string]string{
-		"kubernetes.io/ingress.class": class,
-		"com.openfaas.spec":           string(specJSON),
-	}
+	annotations := function.ObjectMeta.Annotations
```

Comment:
Not keen on assigning an object reference, I would rather see you copy them.

Can you also update any unit tests needed?

Reply:
Changed to clone the annotations map instead of assigning. There aren't any unit tests in this project that I can see.

Reply:
No unit tests you can see? What should we do about that?

https://blog.alexellis.io/golang-writing-unit-tests/